cpp/arcticdb/column_store/chunked_buffer.hpp

```diff
@@ -164,9 +163,12 @@ class ChunkedBufferImpl {
     // the buffer to stay regular sized for as long as possible, which greatly improves random access performance, and
     // will also be beneficial with the slab allocator.
     uint8_t* ensure(size_t requested_size, bool aligned=false) {
-        if (requested_size == 0 || requested_size <= bytes_)
+        if (requested_size != 0 && requested_size <= bytes_)
             return last_block().end();
 
+        if(requested_size == 0)
+            return nullptr;
```

Comment:
This is a bit counterintuitive if there is at least one block

Reply:
I'm trying to keep the branches to a minimum. This is about the null values changes sometimes allocating a column with no data at all, which never used to happen. In that case, we don't want to do any further allocations beyond the column itself

Reply:
Also if you ask for no data, you have to expect not to be able to write to it